Networking/WebSockets/WebSocketProtocol.hh

```diff
@@ -332,28 +337,28 @@ namespace uWS {
             memcpy(dst + headerLength, src, length);
 
             if ( !isServer ) {
-#if 1
-                // overwrites up to 3 bytes outside of the given buffer!
-                //WebSocketProtocol<isServer>::unmaskInplace(dst + headerLength, dst + headerLength + length, mask);
-
-                // this is not optimal
-                std::byte* start = dst + headerLength;
-                std::byte* stop  = start + length;
-                int        i     = 0;
-                while ( start != stop ) { (*start++) ^= mask[i++ % 4]; }
-#else
-                uint32_t mask_i32 = *reinterpret_cast<uint32_t*>(mask);
+                std::byte* start_byte = dst + headerLength;
+                std::byte* end_byte   = start_byte + length;
+                size_t     offset     = 0;
 
-                auto*     start = reinterpret_cast<uint32_t*>(dst + headerLength);
-                uint32_t* end   = start + length / 4;
+                // Handle first x amount of bytes individually until we are aligned with the 4-byte alignment
+                for ( ; (reinterpret_cast<uintptr_t>(start_byte) & 3) != 0 && start_byte != end_byte; ++offset ) {
+                    *start_byte++ ^= mask[offset % 4];
+                }
 
+                // Rotate mask by the offset we reached in the first loop to become memory-aligned
+                std::rotate(mask.begin(), mask.begin() + offset, mask.end());
```

Comment:
Is offset guaranteed to be < 4?

Reply:
With a 4-byte alignment, the maximum mathematically possible offset is 3.  If it is offset by 4, it is once again aligned.

Reply:
I referred to "offset". It advances with *start_byte++ . Should it modulo 4, %= 4 inside std::rotate?

Reply:
It will never be greater than 3, because of the conditions it is set under. It is incremented until start_byte % 4 == 0 (equivalent to start_byte & 3), so it will be 0, 1, 2 or 3.

Reply:
Okay, so, '%' in "mask[offset % 4]" is not necessary.

Reply:
Yeah, you're right. Think it was just leftover from copying the final loop